applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c

```diff
@@ -718,6 +711,19 @@ int unicast_server_enable(le_audio_receive_cb recv_cb)
 
 	channel_assignment_get(&channel);
 
+	if (!IS_ENABLED(CONFIG_BT_CSIP_SET_MEMBER_TEST_SAMPLE_DATA)) {
+		if (!strcmp(CONFIG_BT_SET_IDENTITY_RESOLVING_KEY_DEFAULT,
+			    CONFIG_BT_SET_IDENTITY_RESOLVING_KEY)) {
+			LOG_WRN("CSIP using the default SIRK, must be changed before production");
+		}
+
+		memcpy(csip_param.set_sirk, CONFIG_BT_SET_IDENTITY_RESOLVING_KEY,
+		       MIN(sizeof(CONFIG_BT_SET_IDENTITY_RESOLVING_KEY),
+			   sizeof(csip_param.set_sirk)));
```

Comment:
```suggestion
		memcpy(csip_param.set_sirk, CONFIG_BT_SET_IDENTITY_RESOLVING_KEY,
		       MIN(sizeof(CONFIG_BT_SET_IDENTITY_RESOLVING_KEY),
			   BT_CSIP_SET_SIRK_SIZE));
```
Have you checked what happens if the key is shorter than 16 characters? Do we then also get the '\0'?

I see that broadcast key should use BT_AUDIO_BROADCAST_CODE_SIZE in the MIN as well, but that can probably be done in a separate issue

Reply:
It is demanded that the SIRK is 128 bits -> 16 bytes. This should be documented in the Kconfig and we need a check for it.

Reply:
Yes I have checked for less than 16 characters and the '\0' is included in the key.
So do we only allow SIRKs of 16 bytes?